### PR TITLE
adds 'What to Use lodash for and When to Use It' section to README.md

### PR DESCRIPTION
**Pull request by rtstewart**
In an attempt to help make the lodash repository more approachable to a larger number of users — from informed developers to curious passersby and new and interested visitors like myself — I was curious if you would consider adding a new section to the lodash README that I wrote. It is entitled "What to Use lodash for and When to Use It". The intention is for the new section to provide a brief explanation, in relatively plain and straightforward language, of what lodash is used for and when one should use it.

As I am relatively new to all of this, I hope that I have made this inquiry in a proper and respectful manner. Thank you for your time and consideration. I am interested to hear your thoughts.

### CODE REVIEW

**Comment by jdalton** [comment]
Have you signed [our cla](http://dojofoundation.org/about/cla)?

**Comment by jdalton** [comment]
Are you collaborating with @Gruximillian?

**Comment by rtstewart** [comment]
Hi Mr. Dalton,

 

Yes I just did.

 

Bobby Stewart

**Comment by rtstewart** [comment]
Yes I am. He is a fellow student at https://learn.modern-developer.com/.

File: README.md [hunk 1]
```diff
@@ -35,7 +35,20 @@ Lodash is available in a variety of other builds & module formats.
...
+
+Lodash's thoroughly-tested methods cover gaps that exist in the JavaScript language itself and can be used for:
+
+* manipulating and iterating over arrays, strings, objects and collections;
+* testing and manipulating values;
+* creating composite functions that perform many common and less common programming tasks using syntax that is easily understandable;
+* and [much more](https://lodash.com/docs).
+
+An important feature of lodash is that it is **customizable with regard to build**. You can mix and match individual functions to create a build to suit your exact needs.  
+Loadash allows computations such as **producing unique arrays, intersections of collections, and many others**, to be performed with minimal effort.
```
Comment (jfmengels) [inline README.md]: There is a typo here, `Loadash` --> `Lodash`.

Comment (rtstewart) [inline README.md]: corrected Loadash typo

**Comment by rtstewart** [comment]
Thank you for your kindness gentlemen.

**Comment by jdalton** [comment]
You're welcome! @rtstewart @Gruximillian feel free to contribute in the future too :)

**Comment by Gruximillian** [comment]
Thank you, we were glad to be able to contribute.

**Comment by jdalton** [comment]
If your school is looking for future contributions I'd :heart: help adding more content on our `lodash/fp` usage.

**Comment by rtstewart** [comment]
@jdalton - Thanks for being so welcoming and taking the time with us. I would love to contribute in a more substantive way as I'm sure my partner @Gruximillian would. We have yet to get to the 'meat' of JS programming in the program, so I would think at that time we may be more capable of contributing to this amazing repo in an appropriate and meaningful way. Many of us have some basic JS skills and are now learning some dev tools and skills such as Git/GitHub prior to the substantial JS material. Can you elaborate to point us at any specific or broad ideas or goals with that? You are referring to: https://github.com/lodash/lodash/wiki/lodash-fp-in-browser-usage-example ?

**Comment by jdalton** [comment]
> Can you elaborate to point us at any specific or broad ideas or goals with that?

The `fp` wrapper of `require('lodash/fp')` transforms lodash methods into auto-curried immutable iteratee-first data-last methods. That means instead of
```js
var isEven = function(n) { return n % 2 == 0; };

var array = [1, 2, 3];
_.remove(array, isEven);
// => [2]
console.log(array);
// => [1, 3]
```
it's
```js
var array = [1, 2, 3];
var removeEven = _.remove(isEven);

removeEven(array);
// => [1, 3]
console.log(array);
// => [1, 2, 3]
```
> You are referring to: [lodash/wiki/lodash-fp-in-browser-usage-example](https://github.com/lodash/lodash/wiki/lodash-fp-in-browser-usage-example)?

Yes, the wiki entry will be cleaned up but something there or a similar wiki location would be great.

**Pull request merged**